File: BOJ/C++/1004-어린왕자.cpp
#include <iostream>
#include <cmath>

int main(){
    int loopNum, startX, startY, endX, endY, planetCount, planetX, planetY, planetRadius, count;
    double distance, distance2;
    std::cin >> loopNum;
    for(int i = 0; i <loopNum; i++)
    {
        std::cin >> startX >> startY >> endX >> endY;
        std::cin >> planetCount;
        count = 0;
        for(int k = 0; k < planetCount; k++)
        {
            std:: cin >> planetX >> planetY >> planetRadius;
            distance = pow((startX - planetX), 2) + pow((startY - planetY), 2);
            distance2 = pow((endX - planetX), 2) + pow((endY - planetY), 2);
            planetRadius = pow(planetRadius, 2);
            count += (planetRadius > distance) ^ (planetRadius > distance2);
        }
        std::cout << count << '\n';
    }
    return 0;
}

File: BOJ/Python/1004-어린왕자.py
for x in range(int(input())):
    a = [int(i) for i in input().split()]
    startPos = (a[0], a[1])
    endPos = (a[2], a[3])  
    count = 0
    for _ in range(int(input())):
        planet = [int(i) for i in input().split()]
        distance = (startPos[0] - planet[0])**2 + (startPos[1] - planet[1])**2
        distance2 = (endPos[0] - planet[0])**2 + (endPos[1] - planet[1])**2
        count += (planet[2]**2 > distance) ^ (planet[2]**2 > distance2)
    print(count)